### ISSUE TITLE
[BITV] 9.2.4.3/3.8 - When the "Select file to link to" dialog is closed, the focus is not returned to the "Link file" button but lost (beginning of page). (2)

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/text#4852]
<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-89806fe784-v3-n8

</details>

- [ ] 9.2.4.3/10.8 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-89806fe784-v10-n8

**Comment by susnux** [comment]
Problem:
1. If you click in the menu bar on the link symbol
2. Then on Link to File
3. Then the Filepicker opens
4. Then close the Filepicker without selecting anything

Now the menu is closed and the focus is on the begin of the document.

Instead the focus should stay on the **open** menu, on the "Link to file" action entry.

**Comment by juliushaertl** [comment]
I can no longer reproduce this issue. @susnux Could you quickly confirm? Otherwise I'd be curious if a step is missing.

I noticed a different one that the viewer closes if i press Escape in the file picker. Looks like generally the escape key event is not stopped in NcModal. Proposed a fix for that in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4677

**Comment by susnux** [comment]
It somewhat works now as the focus stays at least on the link menu toggle.

But I am not sure if that is enough as maybe the menu needs to stay open? But at least better :tada:
